### PR TITLE
Added function randomString

### PR DESCRIPTION
**Pull request by axolm**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added a function `randomString` that generates binary string with random bytes (including zero bytes).

### CODE REVIEW

File: src/Functions/randomString.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int TOO_LARGE_STRING_SIZE;
+}
+
+
+/* Generate random string of specified length with fully random bytes(including zero). */
+class FunctionRandomPrintableASCII : public IFunction
```
Comment (qoega) [inline src/Functions/randomString.cpp:20]: Are you going to reimplement https://clickhouse.tech/docs/en/sql-reference/functions/other-functions/#randomascii ? 
Have you checked https://clickhouse.tech/docs/en/sql-reference/table-functions/generate/#generaterandom ?
Probably it is much better to add settings that improve `generateRandom` if you don't have specific cases, where random string should be generated by separate code.

**Comment by alexey-milovidov** [comment]
```
Андрей Некрашевич, [08.05.20 11:29]
Привет! А я правильно понимаю, что мне теперь не нужно так (https://github.com/ClickHouse/ClickHouse/pull/10733/files) копипастить эти (https://clickhouse.tech/docs/ru/whats-new/extended-roadmap/#funktsii-dlia-fuzzing) функции, потому что уже есть generateRandom, и сейчас нужно написать те, которые инвертят битики. @qoega заставил сомневаться.

Alexey Milovidov, [09.05.20 17:30]
Одна из функций полностью потеряла актуальность:
randomStringASCII уже реализовали в виде randomPrintableASCII.

Другие актуальны, но слегка меньше:
randomString
randomFixedString
randomStringUTF8

потому что похожая функциональность реализована в табличной функции generateRandom... но не совсем такая, какая требуется от этих функций. Поэтому они остаются актуальны.

fuzzBits
fuzzBytes
- а вот это как раз очень актуально.

Alexey Milovidov, [09.05.20 17:31]
Fuzzing лексера и парсера запросов; кодеков и форматов
- а это вообще отдельная часть задачи, гораздо более глубокая. Я могу дать логин человека, с кем проконсультироваться, что сделать в первую очередь.

Alexey Milovidov, [09.05.20 17:32]
randomString
randomFixedString
- просто равномерно случайные байты, в том числе нулевые байты.

Alexey Milovidov, [09.05.20 17:34]
randomStringUTF8
- равномерно случайный корректный UTF-8. Здесь чуть посложнее - нужны символы не только из basic multilingual plane; но можно использовать unassigned code points для простоты. То есть, какой-то диапазон кодовых точек unicode.

Андрей Некрашевич, [09.05.20 17:36]
[In reply to Alexey Milovidov]
спасибо за ответ
дай логин, пожалуйста

Alexey Milovidov, [09.05.20 17:37]
@kyprizel

Alexey Milovidov, [09.05.20 17:37]
Сейчас его предупрежу, что ты придёшь спросить про фаззинг...

Alexey Milovidov, [09.05.20 17:38]
Готово :)

Андрей Некрашевич, [09.05.20 17:38]
благодарю)

Alexey Milovidov, [09.05.20 17:39]
Это про вторую часть задачи (которая более глубокая) - про автоматизированный фаззинг компонентов системы.
```

File: src/Functions/randomString.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            IColumn::Offset next_offset = offset + length + 1;
+            data_to.resize(next_offset);
+            offsets_to[row_num] = next_offset;
+
+            auto * data_to_ptr = data_to.data(); // avoid assert on array indexing after end
+            for (size_t pos = offset, end = offset + length; pos < end;
+                 pos += 8) // We have padding in column buffers that we can overwrite.
+            {
+                UInt64 rand = thread_local_rng();
+                data_to_ptr[pos] = rand;
```
Comment (alexey-milovidov) [inline src/Functions/randomString.cpp:81]: This is slightly incorrect.

**Comment by alexey-milovidov** [comment]
Need to add performance test.

**Comment by alexey-milovidov** [comment]
Let's finish this PR and proceed to other functions.

**Pull request merged**